GSA-TTS/FAC-Frontend
# Implement access denial

Implementation of GSA-TTS/FAC#401

If a user isn't logged in, they'll be redirected to the homepage and shown this alert:

<img width="377" alt="image" src="https://user-images.githubusercontent.com/6290/182655330-199f4ae8-77fc-491a-bf67-0ea51633729f.png">

Similarly, if a logged-in user tries to go to any page associated with a particular audit submission (any page with a `reportId=[MY SUBMISSION ID]` query param, they will get the same redirect and the same error message.

## QA Note

🦅 [Federalist preview](https://federalist-9a617ff8-042d-4076-9581-bb999f9c6639.app.cloud.gov/preview/gsa-tts/fac-frontend/mh/access-denial-401/)

To test this, you'll need to have gone through the pre-SAC flow at least once so that you'll have at least one report ID associated with your user. Once you've done that you'll need to:

1. Get the `report_id` for one of the audits associated with your user account;
2. Since we haven't yet implemented any pages that use a `reportId` query param, after logging in, you can go to any page after the home page and add that param yourself (i.e. append `?reportId=[YOUR REPORT ID]` to the URL), hit return, and observe that the page reloads normally; then
3. To see the unhappy path, change the query param value to something else (`foo`, etc.), hit return again, and be redirected home.

Or just see the following video:

https://user-images.githubusercontent.com/6290/187265471-b5819692-df55-43c0-88a5-23878c462a61.mov
